### DISCUSSION TITLE
Netbox as source of truth

### DISCUSSION THREAD

**Discussion by robertmain** [netbox-community/netbox#13853] [General]
First off I love netbox and have my network (and one I manage for a church) mapped out in NetBox and it makes a TON of stuff easier. That said, there's something that's been bothering me and that's Netbox's intended use as a source of truth. 

My understanding is that using config templates one would use something like ansible to propogade the changes in NetBox out to the network. I'd assume this is done in a somewhat immutable manner meaning that (presumably) the entire configuration state of said device needs to be representable in the config file applied by ansible.

For things like interface names, IPs and VLANs this is fine. But it becomes trickier for things like DHCP/IGMP snooping settings on a switch, NAT configuration on a router etc. 

Given Netbox's usage as a L2/L2.5/L3 representation of a network, how does one represent these changes in a netbox config template?

**Comment by candlerb** [comment]
If you want to build the complete template from Netbox then indeed you'll need all the parameters for the template in Netbox. This can be achieved using [configuration contexts](https://docs.netbox.dev/en/stable/models/extras/configcontext/), [custom fields](https://docs.netbox.dev/en/stable/models/extras/customfield/), and/or tags and roles (depending on how you define them).

**Reply by DanSheps** [reply]
I plan on creating a plugin to add some more l2 features to interfaces to replace custom fields.
